synfig-studio/src/gui/docks/dock_toolbox.cpp

```diff
@@ -231,6 +234,28 @@ Dock_Toolbox::add_state(const Smach::state_base *state)
 
 	refresh();
 }
+void Dock_Toolbox::new_layer(){
+	Gtk::RadioToolButton *tool_button = manage(new Gtk::RadioToolButton());
+	tool_button->set_group(radio_tool_button_group);
+	tool_button->set_related_action(action_new_layer);
+
+	tool_button->show();
+
+	tool_item_group->insert(*tool_button);
+	tool_item_group->show_all();
+
+
+	refresh();
+
+}
+void Dock_Toolbox::popup_add_layer_menu()
+{
+	if (!action_new_layer->is_sensitive())
+		return;
+	Gtk::Menu* menu = dynamic_cast<Gtk::Menu*>(App::ui_manager()->get_widget("/popup-layer-new"));
```

Comment:
If you only need the value inside the `if` condition, it's better to use something like this (to reduce variable scope):
```
if (auto menu = dynamic_cast<Gtk::Menu*>(App::ui_manager()->get_widget("/popup-layer-new")))
	menu->popup(0, gtk_get_current_event_time());
```
It also shows that we don't need this variable anymore.